phoenix-core-server/src/main/java/org/apache/phoenix/coprocessor/UncoveredIndexRegionScanner.java

```diff
@@ -299,21 +299,28 @@ private boolean verifyIndexRowAndRepairIfNecessary(Result dataRow, byte[] indexR
         for (Cell cell : dataRow.rawCells()) {
             put.add(cell);
         }
-        if (indexMaintainer.checkIndexRow(indexRowKey, put)) {
-            if (IndexUtil.getMaxTimestamp(put) != indexTimestamp) {
-                Mutation[] mutations;
-                Put indexPut = new Put(indexRowKey);
-                indexPut.addColumn(emptyCF, emptyCQ, indexTimestamp, QueryConstants.VERIFIED_BYTES);
-                if ((EnvironmentEdgeManager.currentTimeMillis() - indexTimestamp) > ageThreshold) {
-                    Delete indexDelete = indexMaintainer.buildRowDeleteMutation(indexRowKey,
-                            IndexMaintainer.DeleteType.SINGLE_VERSION, indexTimestamp);
-                    mutations = new Mutation[]{indexPut, indexDelete};
-                } else {
-                    mutations = new Mutation[]{indexPut};
+        if (indexMaintainer.isCDCIndex()) {
+            if (indexMaintainer.checkIndexRow(indexRowKey, dataRow.getRow(), put, viewConstants)
```

Comment:
Do we really need to call checkIndexRow as it will always return true since the index row does not include any non PK data table columns?